base_classes/NXobject.nxdl.xml

```diff
@@ -30,5 +30,164 @@
 		This is the base object of NeXus
 	</doc>
         <!--attribute name="name"><doc>name of instance</doc></attribute-->
-</definition>
+	<field name="identifierNAME" type="NX_CHAR" nameType="partial">
+		<doc>
+			An identifier for a (persistent) resource.
 
+			An identifier, provided by some authority, that has been assigned to the
+			real-world object described by this ``NXobject``. To be useful, the identifier
+			must not be reassigned to a different real-world object. It is typical for
+			there to be some mechanism to resolve an identifier, obtaining metadata
+			about the object. Identifiers for which some guarantees exist regarding
+			this resolution process are called persistent identifiers.
+			Persistent identifiers are also known as PIDs.
+		</doc>
+		<attribute name="type" type="NX_CHAR">
+			<doc>The type of identifier used.</doc>
+			<enumeration>
+				<item value="ARK">
+					<doc>
+						Archival Resource Key.
+						
+						An ARK is a Uniform Resource Identifier (URI) designed to support long-term access to a variety of information objects.
+						
+						Syntax: https://NMA/ark:/NAAN/Name[Qualifier]. Brackets indicate optional elements.
+						Example: https://example.org/ark:/12345/abcde
+					</doc>
+				</item>
+				<item value="DOI">
+					<doc>
+						Digital Object Identifier.
+						
+						A DOI is a unique alphanumeric string used to identify digital content. It consists of a prefix and a suffix, separated by a slash.
+						
+						Syntax: 10.XXXX/XXXXXX
+						Example: 10.1107/S1600576714027575
+					</doc>
+				</item>
+				<item value="Hdl">
+					<doc>
+						A handle is a unique identifier that consists of a prefix indicating the naming authority and a suffix representing the local name 
+						of a resource, cts. 
+						
+						A handle is a unique identifier used to facilitate the identification and management of digital objects. It is composed of a prefix that 
+						indicates the naming authority and a suffix that specifies the resource's local name. 
+						
+						This refers specifically to an ID in the Handle system operated by the Corporation for National Research Initiatives (CNRI).
+						
+						Syntax: hdl:prefix/identifier
+						Example: hdl:123456789/abc123
+					</doc>
+				</item>
+				<item value="IGSN">
+					<doc>
+						International Generic Sample Number.
+						
+						The IGSN is a unique identifier assigned to a specific sample or specimen in the context of scientific research.
+						
+						Syntax: https://igsn.org/{IGSN-ID}
```

Comment:
I'm not convinced this syntax is correct.

Per [the Wikipedia article](https://en.wikipedia.org/wiki/International_Generic_Sample_Number#History), IGSNs are now issued by DataCite.

Also, from this wikipedia article, an example of a DateCite issued IGSN is `10.58052/SSH000SUA`.  Note that this is also a DOI.

Per [the partnership agreement](https://datacite.org/blog/igsn-and-datacite-partnership/):

    Existing IGSN ID handles will now be registered IGSN ID
    DOIs and the handles aliased to the DOIs to ensure that
    these continue to resolve. 

I take this to mean that there are now DataCite-issued DOIs for all IGSNs, including those historical IGSNs issued before the partnership.

Therefore, I believe NeXus can use the same syntax for DOIs and IGSNs.

Reply:
I agree, this was the older syntax. I added some text describing this change:
```
Since 2021, IGSNs are issued by DataCite, meaning that hat there are now DataCite-issued DOIs for all IGSNs, including those historical IGSNs issued beforehand. Therefore, the syntax is the same as for DOIs.

Syntax: 10.XXXX/XXXXXX
Example: 10.1107/S1600576714027575
```

Reply:
`hat` to `that`

Reply:
This was already addressed in the NXDL file itself.